frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -206,14 +203,33 @@ export const getDynamicEventingChannelWatchers = (namespace: string) => {
     return acc;
   }, {});
 };
+export const useChannelResourcesList = (): string[] => {
+  const [modelRefs, setModelRefs] = useSafetyFirst<string[]>([]);
+  useEffect(() => {
+    if (eventSourceData.eventSourceChannels.length === 0) {
+      fetchChannelsCrd()
+        .then((data) => {
+          setModelRefs(data.map((model: K8sKind) => referenceForModel(model)));
+        })
+        .catch((err) => {
+          setModelRefs([]);
+          // eslint-disable-next-line no-console
+          console.warn('Error fetching CRDs for dynamic channel model refs', err);
+        });
+    } else {
+      setModelRefs(
+        eventSourceData.eventSourceChannels.map((model: K8sKind) => referenceForModel(model)),
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I think it's okay to keep the dependency here 
```suggestion
  }, [setModelRefs]);
```

Reply:
Updated.